juju/model.py

```diff
@@ -1005,18 +1008,36 @@ def debug_log(
                 # haven't made it yet we'll need to wait on them to be added
                 await asyncio.gather(*[
                     asyncio.ensure_future(
-                        self._wait_for_new('application', app_name))
+                        self._wait_for_new('application', app_name),
+                        loop=self.loop)
                     for app_name in pending_apps
-                ])
+                ], loop=self.loop)
             return [app for name, app in self.applications.items()
                     if name in handler.applications]
         else:
-            log.debug(
-                'Deploying %s', entity_id)
-
             if not is_local:
+                if not application_name:
```

Comment:
It should be covered by https://github.com/juju/python-libjuju/pull/63/files#diff-2e7ca22cc563b1b07795f08e3ee7cf42R103 shouldn't it?